File: KuantumKaosYönetimi.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});


const soruSor = (soru) => new Promise(resolve => rl.question(soru, resolve));


class KuantumCokusuException extends Error {
    constructor(id) {
        super(`KRİTİK HATA: Nesne ${id} stabilitesini kaybetti!`);
        this.name = "KuantumCokusuException";
    }
}


class KuantumNesnesi {
    constructor(id, stabilite, tehlikeSeviyesi) {
        if (this.constructor === KuantumNesnesi) {
            throw new Error("Abstract sınıf new ile oluşturulamaz.");
        }
        this._id = id;
        this._tehlikeSeviyesi = tehlikeSeviyesi;
        this.setStabilite(stabilite);
    }

    getId() { return this._id; }

    getStabilite() { return this._stabilite; }

    setStabilite(val) {
        if (val > 100) {
            this._stabilite = 100;
        } else if (val <= 0) {
            this._stabilite = 0;
            throw new KuantumCokusuException(this._id);
        } else {
            this._stabilite = val;
        }
    }

    analizEt() {
        throw new Error("Bu metod override edilmelidir.");
    }

    durumBilgisi() {
        return `ID: ${this._id} | Stabilite: %${this._stabilite} | Tehlike: ${this._tehlikeSeviyesi}`;
    }
}


class VeriPaketi extends KuantumNesnesi {
    constructor(id) {
        super(id, 100, 1);
    }

    analizEt() {
        this.setStabilite(this.getStabilite() - 5);
        console.log(">> Veri içeriği okundu.");
    }
}

class KaranlikMadde extends KuantumNesnesi {
    constructor(id) {
        super(id, 100, 7);
    }

    analizEt() {
        this.setStabilite(this.getStabilite() - 15);
        console.log(">> Karanlık madde analiz edildi.");
    }

    
    acilDurumSogutmasi() {
        try {
            this.setStabilite(this.getStabilite() + 50);
            console.log(">> Soğutma başarılı.");
        } catch (e) {}
    }
}

class AntiMadde extends KuantumNesnesi {
    constructor(id) {
        super(id, 100, 10);
    }

    analizEt() {
        this.setStabilite(this.getStabilite() - 25);
        console.log(">> UYARI: Evrenin dokusu titriyor...");
    }

    acilDurumSogutmasi() {
        try {
            this.setStabilite(this.getStabilite() + 50);
            console.log(">> Anti-Madde stabilize edildi.");
        } catch (e) {}
    }
}


async function main() {
    let envanter = [];
    let sayac = 1;
    let sistemCalisiyor = true;

    console.log("--- KUANTUM AMBARI ---");

    while (sistemCalisiyor) {
        console.log("\n1. Yeni Nesne Ekle");
        console.log("2. Tüm Envanteri Listele");
        console.log("3. Nesneyi Analiz Et");
        console.log("4. Acil Durum Soğutması Yap");
        console.log("5. Çıkış");

        const secim = await soruSor("Seçiminiz: ");

        try {
            switch (secim) {
                case '1':
                    const rand = Math.floor(Math.random() * 3);
                    let yeniNesne;
                    const id = `NESNE-${1000 + sayac++}`;
                    if (rand === 0) yeniNesne = new VeriPaketi(id);

                    else if (rand === 1) yeniNesne = new KaranlikMadde(id);

                    else yeniNesne = new AntiMadde(id);
                    
                    envanter.push(yeniNesne);
                    console.log(`Eklendi: ${yeniNesne.constructor.name}`);
                    break;

                case '2':

                    if (envanter.length === 0) console.log("Depo boş.");
                    envanter.forEach(n => {
                        console.log(`${n.durumBilgisi()} [${n.constructor.name}]`);
                    });
                    break;

                case '3':

                    const aId = await soruSor("ID giriniz: ");
                    const aNesne = envanter.find(n => n.getId() === aId);
                    if (aNesne) {
                        aNesne.analizEt(); 
                    } else {
                        console.log("Nesne bulunamadı.");
                    }
                    break;

                case '4':

                    const sId = await soruSor("ID giriniz: ");
                    const sNesne = envanter.find(n => n.getId() === sId);
                    if (sNesne) {
                        
                        if (typeof sNesne.acilDurumSogutmasi === 'function') {
                            sNesne.acilDurumSogutmasi();

                        } else {
                            console.log("HATA: Bu nesne soğutulamaz!");
                        } }
                         else {

                        console.log("Nesne bulunamadı.");
                    }
                    break;

                case '5':
                    sistemCalisiyor = false;
                    console.log("Çıkış yapıldı.");
                    break;

                default:
                    console.log("Geçersiz seçim.");
            }
        } catch (e) {
            if (e.name === "KuantumCokusuException") {

                console.log("SİSTEM ÇÖKTÜ! TAHLİYE BAŞLATILIYOR...");
                console.log(e.message);
                sistemCalisiyor = false;
           
            } else {
                console.log("Beklenmedik bir hata oluştu: " + e);
            }
        }
    }
    rl.close();
}

main();

File: KuantumKaosYönetimi.py
import random
from abc import ABC, abstractmethod


class KuantumCokusuException(Exception):
    def __init__(self, nesne_id):
        self.message = f"KRİTİK HATA: Nesne {nesne_id} patladı! Sistem Çöktü."
        super().__init__(self.message)


class IKritik(ABC):
    @abstractmethod
    def acil_durum_sogutmasi(self):
        pass


class KuantumNesnesi(ABC):
    def __init__(self, id, stabilite, tehlike_seviyesi):
        self._id = id
        self._tehlike_seviyesi = tehlike_seviyesi
        self._stabilite = 100
        self.stabilite = stabilite 

    @property
    def id(self):
        return self._id

    @property
    def stabilite(self):
        return self._stabilite

    @stabilite.setter
    def stabilite(self, value):
        if value > 100:
            self._stabilite = 100
        elif value <= 0:
            self._stabilite = 0
            
            raise KuantumCokusuException(self._id)
        else:
            self._stabilite = value

    @abstractmethod
    def analiz_et(self):
        pass

    def durum_bilgisi(self):
        return f"ID: {self._id} | Stabilite: %{self._stabilite} | Tehlike: {self._tehlike_seviyesi}"


class VeriPaketi(KuantumNesnesi):
    def __init__(self, id):
        super().__init__(id, 100, 1)

    def analiz_et(self):
        self.stabilite -= 5
        print(">> Veri içeriği okundu.")

class KaranlikMadde(KuantumNesnesi, IKritik):
    def __init__(self, id):
        super().__init__(id, 100, 7)

    def analiz_et(self):
        self.stabilite -= 15
        print(">> Karanlık madde analiz edildi.")

    def acil_durum_sogutmasi(self):
        try:
            self.stabilite += 50
            print(f">> {self.id} soğutuldu. Yeni Stabilite: {self.stabilite}")
        except KuantumCokusuException:
            pass 

class AntiMadde(KuantumNesnesi, IKritik):
    def __init__(self, id):
        super().__init__(id, 100, 10)

    def analiz_et(self):
        self.stabilite -= 25
        print(">> UYARI: Evrenin dokusu titriyor...")

    def acil_durum_sogutmasi(self):
        try:
            self.stabilite += 50
            print(f">> {self.id} (Anti-Madde) soğutuldu.")
        except KuantumCokusuException:
            pass


def main():
    envanter = []
    sayac = 1
    sistem_calisiyor = True

    print("--- KUANTUM AMBARI ---")

    while sistem_calisiyor:
        print("\n1. Yeni Nesne Ekle")
        print("2. Tüm Envanteri Listele")
        print("3. Nesneyi Analiz Et")
        print("4. Acil Durum Soğutması Yap")
        print("5. Çıkış")
        secim = input("Seçiminiz: ")

        try:
            if secim == "1":
                yeni_id = f"NESNE-{1000 + sayac}"
                sayac += 1
                tip = random.choice([VeriPaketi, KaranlikMadde, AntiMadde])
                nesne = tip(yeni_id)
                envanter.append(nesne)
                print(f"Eklendi: {type(nesne).__name__}")

            elif secim == "2":
                for n in envanter:
                    print(f"{n.durum_bilgisi()} [{type(n).__name__}]")

            elif secim == "3":
                id_gir = input("ID Giriniz: ")
                bulundu = False
                for n in envanter:
                    if n.id == id_gir:
                        n.analiz_et()
                        bulundu = True
                        break
                if not bulundu: print("Nesne bulunamadı.")

            elif secim == "4":
                id_gir = input("ID Giriniz: ")
                bulundu = False
                for n in envanter:
                    if n.id == id_gir:
                        
                        if isinstance(n, IKritik):
                            n.acil_durum_sogutmasi()
                        else:
                            print("HATA: Bu nesne soğutulamaz!")
                        bulundu = True
                        break
                if not bulundu: print("Nesne bulunamadı.")

            elif secim == "5":
                sistem_calisiyor = False
                print("Çıkış yapıldı.")
            
        except KuantumCokusuException as e:
          
            print("SİSTEM ÇÖKTÜ! TAHLİYE BAŞLATILIYOR...")
            print(e)
        
            sistem_calisiyor = False

if __name__ == "__main__":
    main()